lib/std/Thread/StaticResetEvent.zig

```diff
@@ -160,27 +160,28 @@ pub const AtomicEvent = struct {
     }
 
     pub fn reset(ev: *AtomicEvent) void {
-        @atomicStore(u32, &ev.waiters, 0, .Monotonic);
+        ev.waiters.store(0, .Monotonic);
     }
 
     pub const Futex = switch (builtin.os.tag) {
         .windows => WindowsFutex,
         .linux => LinuxFutex,
+        .macos => std.Thread.Futex,
```

Comment:
Futex.wait() usage in `timedWait()` is expected to not have spurious wakeups but `std.Thread.Futex`'s API does. Convert it into a loop to account for this. See the following pseudo code

```zig
struct:
  wake(waiters, count):
    Futex.wake(waiters, count)
    
  wait(waiters, timeout):
    deadline = timeout != null ? nanoTimestamp() : null
    while true:
      w = waiters.load(Acquire)
      if w == WAKE: return
      Futex.wait(waiters, w, blk:
        _ = timeout orelse break null
        now = nanoTimestamp()
        if now >= deadline: return error.TimedOut
        break deadline - now
```

Not really supposed to use `nanoTimestamp()` for tracking elapsed time, but whatever. Also might as well use that pseudo code for all platforms not just macos (since std.Thread.Futex is portable and doesn't spin)